core/src/test/java/org/infinispan/api/CacheAPITest.java

```diff
@@ -271,4 +272,113 @@ public void testPutIfAbsentAfterRemoveInTx() throws SystemException, NotSupporte
 
       assertEquals(old_value, cache.get(key));
    }
+
+   public void testSizeInExplicitTxWithNonExistent() throws SystemException, NotSupportedException {
+      assertEquals(0, cache.size());
+      cache.put("k", "v");
+
+      TransactionManager tm1 = TestingUtil.getTransactionManager(cache);
+      tm1.begin();
+      try {
+         cache.get("no-exist");
```

Comment:
assertNull? And the withTx... though I know you've just moved this code from another test.